openapi.yml

```diff
@@ -0,0 +1,475 @@
+        - name: job-id
+          in: path
+          description: ID of the ProcessorJob
+          schema: {type: string}
+          required: true
+      responses:
+        '200':
+          description: Return Log
+          content:
+            'text/plain': {}
+        '404': {content: {}, description: 'ProcessorJob not found'}
+    post:
+      tags: ['processing']
+      operationId: logProcessor
+      parameters:
+        - name: executable
+          in: path
+          description: LogEntry to log
+          schema: {$ref: '#/components/schemas/LogEntry'}
+          required: true
+        - name: job-id
+          in: path
+          description: ID of the ProcessorJob
+          schema: {type: string}
+          required: true
+      responses:
+        '200':
+          description: Return Log
+          content:
+            'text/plain': {}
+        '404': {content: {}, description: 'ProcessorJob not found'}
+
+  '/workflow':
+    post:
+      tags: ['workflow', 'discovery']
+      operationId: postWorkflow
+      requestBody:
+        description: 'Register a new workflow'
+        content:
+          'text/vnd.ocrd+sh': {}
+        required: true
+      responses:
+        '200':
+          description: Created a new OCR-D workflow
+          content: {application/json: {schema: {$ref: '#/components/schemas/Workflow'}}}
+
+  '/workflow/{workflow-id}':
+    put:
+      tags: ['workflow']
+      operationId: putWorkflow
+      parameters:
+        - name: workflow-id
+          in: path
+          description: ID of the Workflow
+          schema: {type: string}
+          required: true
+      requestBody:
+        description: 'Replace existing or create new workflow'
+        content:
+          'text/vnd.ocrd+sh': {}
+        required: true
+      responses:
+        '200':
+          description: Created a new OCR-D workflow
+          content: {application/json: {schema: {$ref: '#/components/schemas/Workflow'}}}
+    get:
+      tags: ['workflow', 'discovery']
+      operationId: getWorkflow
+      parameters:
+        - name: workflow-id
+          in: path
+          description: ID of the Workflow
+          schema: {type: string}
+          required: true
+      responses:
+        '200':
+          description: Return ProcessorJob
+          content:
+            application/json: {schema: {$ref: '#/components/schemas/Workflow'}}
+            application/vnd.ocrd+zip: {}
+        '404': {content: {}, description: 'Workflow not available'}
+    post:
+      tags: ['workflow']
+      operationId: runWorkflow
+      parameters:
+        - name: workflow-id
+          in: path
+          description: ID of the Workflow
+          schema: {type: string}
+          required: true
+      responses:
+        '200':
+          description: Return WorkflowJob
+          content:
+            application/json: {schema: {$ref: '#/components/schemas/WorkflowJob'}}
+
+  '/workflow/{workflow-id}/{job-id}':
```

Comment:
To avoid that problem altogether, one could use UUID. Otherwise, I'd say every counter should be endpoint specific and the `ID` should be the full URL, including the portion referring to a specific processor or workflow to disambiguate

Reply:
Is this something we have to handle via the schema or should it be stated in the specs (cf. #195 )?

Reply:
I think this is an implementation detail and up to the implementers, how they handle IDs. We can certainly recommend something ("Use endpoint-specific counters for IDs" or "Use UUID for IDs"?).

Reply:
Okido. Let's talk about it during the course of #195, then.